misc/Kconfig

```diff
@@ -111,6 +111,13 @@ config CHECK_LINK_MAP
 	help
 	  Run a linker address generation validity checker at the end of the
 	  build.
+
+config LIB_CPLUSPLUS
```

Comment:
can we please move all cpp related Kconfig options to subsys/cpp/Kconfig? misc/Kconfig seems to be odd when we have all the code in subsys/cpp already.

Reply:
> misc/Kconfig seems to be odd when we have all the code in subsys/cpp already.

It's not less odd that we have lib/libc/, and then suddenly subsys/cpp/ . I would never guess to look for it there.